cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -51,11 +49,14 @@ func NewCollectorProxy(o *Options, logger *zap.Logger) (*ProxyBuilder, error) {
 		r.InitialAddrs(resolvedAddrs)
 		conn, _ = grpc.Dial(r.Scheme()+":///round_robin", grpc.WithInsecure(), grpc.WithBalancerName(roundrobin.Name))
 	} else {
+		// It does not return error if the collector is not running
+		// a way to fail immediately is to call WithBlock and WithTimeout
```

Comment:
I don't think we want to fail if collector is not running (maybe it's being re-deployed), as it might kill the agent. Preferably components should stay up when dependencies are down, but report metrics.

Reply:
so I would remove this comment completely